src/index.ts

```diff
@@ -17,7 +17,17 @@ const getFathom = (): Fathom => {
     });
 };
 
-export const load = (url = '//cdn.usefathom.com/tracker.js'): void => {
+// refer to https://usefathom.com/support/tracking-advanced
+export type LoadOptions = {
+  auto?: boolean;
+  honorDNT?: boolean;
+  canonical?: boolean;
+  includedDomains?: string[];
+  excludedDomains?: string[];
+  spa?: 'auto' | 'history' | 'hash';
+}
+
+export const load = (url = '//cdn.usefathom.com/tracker.js', opts?: LoadOptions): void => {
```

Comment:
The tracker now looks for the `site` property on the script tag node to know which site is being tracked. I think it makes sense to:

1. Make `siteId` the first (required) positional argument on the `load` function, now that script loading and setting the site id are coupled, and
2. Move `url` to the options object.

```js highlight=2
export type LoadOptions = {
  url?: string;
  auto?: boolean;
  honorDNT?: boolean;
  canonical?: boolean;
  includedDomains?: string[];
  excludedDomains?: string[];
  spa?: 'auto' | 'history' | 'hash';
}

export const load = (siteId, opts?: LoadOptions): void => {
  window.fathom =
    window.fathom ||
    function() {
      (window.fathom.q = window.fathom.q || []).push(arguments);
    };

  let tracker = document.createElement('script');
  let firstScript = document.getElementsByTagName('script')[0];

  tracker.async = true;
  tracker.setAttribute('site', siteId);
  tracker.src = opts && opts.url ? opts.url : 'https://cdn.usefathom.com/script.js;
  tracker.id = 'fathom-script';

  if (opts) {
    if (opts.auto !== undefined) tracker.setAttribute('auto', `${opts.auto}`);
    if (opts.honorDNT !== undefined) tracker.setAttribute('honor-dnt', `${opts.honorDNT}`);
    if (opts.canonical !== undefined) tracker.setAttribute('canonical', `${opts.canonical}`);
    if (opts.includedDomains) tracker.setAttribute('included-domains', opts.includedDomains.join(','));
    if (opts.excludedDomains) tracker.setAttribute('excluded-domains', opts.excludedDomains.join(','));
    if (opts.spa) tracker.setAttribute('spa', opts.spa);
  }

  firstScript.parentNode.insertBefore(tracker, firstScript);
}
```

Reply:
I was thinking this too - however, I didn't want to introduce a backwards-incompatible change for other users of this client. I think the best way forward might be to:

* land this PR's `load()` parameters as-is
* introduce a `v3` further down the line with the parameters swapped

Reply:
In chatting with Jack from Fathom, I believe these new options only work for the new script, which is backwards-incompatible (no more async queue, looks for `site` on the script tag).

Reply:
Is there a script other than `cdn.usefathom.com/tracker.js`? The [documentation](https://usefathom.com/support/tracking) does not seem to indicate a new backwards-incompatible script is available (though maybe I've missed something?)

Either way, I'm specifically referring to backwards-incompatibility in users of your package `fathom-client` - if they are using `fathom.load(url)`, I don't want to swap out that parameter unexpectedly :)

Reply:
Yes, `cdn.usefathom.com/script.js`:

<img width="731" alt="image" src="https://user-images.githubusercontent.com/341387/82250425-0d9b5980-9911-11ea-98b2-a3455c487669.png">

I'm pretty sure the new options we are adding here are only processed by `script.js` (not `tracker.js`). I'd be inclined to just cut to v3 here.

Reply:
Ah, I completely missed that - gotcha, will make the changes

Reply:
See 36fc3e0